dataRetriever/txpool/mempoolHost.go

```diff
@@ -0,0 +1,112 @@
+package txpool
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/process"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+	"github.com/multiversx/mx-chain-vm-common-go/parsers"
+)
+
+type argsMempoolHost struct {
+	txGasHandler txGasHandler
+	marshalizer  marshal.Marshalizer
+}
+
+type mempoolHost struct {
+	txGasHandler        txGasHandler
+	callArgumentsParser process.CallArgumentsParser
+	esdtTransferParser  vmcommon.ESDTTransferParser
+}
+
+func newMempoolHost(args argsMempoolHost) (*mempoolHost, error) {
+	if check.IfNil(args.txGasHandler) {
+		return nil, dataRetriever.ErrNilTxGasHandler
+	}
+	if check.IfNil(args.marshalizer) {
+		return nil, dataRetriever.ErrNilMarshalizer
+	}
```

Comment:
Somehow superfluous checks (dependencies are verified one layer above, as well), should I remove them?